src/trio/_core/_asyncgens.py

```diff
@@ -230,7 +230,9 @@ async def _finalize_one(
             # with an exception, not even a Cancelled. The inside
             # is cancelled so there's no deadlock risk.
             with _core.CancelScope(shield=True) as cancel_scope:
-                cancel_scope.cancel()
+                cancel_scope.cancel(
+                    reason="internal trio cancellation during finalization"
```

Comment:
```suggestion
                    reason="disallow async work when closing async generators during trio shutdown"
```
it's a mouthful, but I want to be clear about what's going on.